src/dispatch/database/service.py

```diff
@@ -437,6 +440,12 @@ def create_sort_spec(model, sort_by, descending):
             direction = "desc" if direction else "asc"
 
             # we have a complex field, we may need to join
+            try:
+                new_field = json.loads(field)
+                field = new_field.get("key", "")
+            except json.JSONDecodeError:
+                pass
```

Comment:
I think I understand this to mean that sometimes the field will be json but other times not and that is _expected_. It might be good to add a comment to that effect and then explain the two scenarios with an example.